### PR TITLE
Wrote docstings for pymc4.utils

### LINKED ISSUE #254: Write pymc4.utils docstrings
The pymc4/utils.py module is mostly undocumented. We have to fix this in order to solve #142.

### PR DESCRIPTION
**Pull request by lucianopaz**
Closes #254.

### CODE REVIEW

File: pymc4/utils.py [hunk 1]
```diff
@@ -105,57 +118,159 @@ class NameParts:
...
+    def is_valid_untransformed_name(cls, name: str) -> bool:
+        """Test if a name can be used as an untransformed random variable.
+
+        This function attempts to test if the supplied name, by accident,
+        matches the naming pattern used for auto transformed random variables.
+        If it does not, it is assumed to be a potentially valid name.
+
+        Parameters
+        ----------
+        cls : NameParts
```
Comment (ferrine) [inline pymc4/utils.py:130]: It's not a part of the signature, so just keep `name` in the Parameters section

File: pymc4/utils.py [hunk 2]
```diff
@@ -105,57 +118,159 @@ class NameParts:
...
         return match is not None and match["transform"] is None
 
     @classmethod
-    def is_valid_name(cls, name):
+    def is_valid_name(cls, name: str) -> bool:
+        """Test if a name doesn't contain forbidden symbols.
+
+        Parameters
+        ----------
+        cls : NameParts
```
Comment (ferrine) [inline pymc4/utils.py:151]: same goes  here

File: pymc4/utils.py [hunk 3]
```diff
@@ -105,57 +118,159 @@ class NameParts:
...
+            otherwise.
+
+        """
         match = cls.NAME_RE.match(name)
         return match is not None
 
-    def __init__(self, path, transform_name, untransformed_name):
+    def __init__(
+        self,
+        path: Union[List[str], Tuple[str]],
```
Comment (ferrine) [inline pymc4/utils.py:168]: Union[List[str], Tuple[str]] -> Sequence[str]

**Comment by lucianopaz** [comment]
Thanks for the review @ferrine. I addressed your comments and everything should be ready to merge now

**Pull request merged**